src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXml.cs

```diff
@@ -117,11 +118,11 @@ internal byte[] GetBytes()
             return utf8.GetBytes(sb.ToString());
         }
 
-        internal byte[] GetDigestedBytes(HashAlgorithm hash)
+        internal byte[]? GetDigestedBytes(HashAlgorithm hash)
         {
             _c14nDoc.WriteHash(hash, DocPosition.BeforeRootElement, _ancMgr);
             hash.TransformFinalBlock(Array.Empty<byte>(), 0, 0);
-            byte[] res = (byte[])hash.Hash.Clone();
+            byte[]? res = (byte[])hash.Hash!.Clone();
```

Comment:
Clone only produces null if the input was null.  `hash.Hash` won't be null since TransformFinalBlock was called, and so Clone() won't return null.  We can get rid of some `?`s here and probably cascade a bit.

Reply:
Actually, this is the built-in `Array.Clone()`.  It can't ever return null, since it needed a non-null instance... so why is `res` declared as may-be null?  (I thought this was hitting an extension method that accepted null)